nix/haskell.nix

```diff
@@ -48,6 +48,7 @@ let
           ghcid
           haskell-language-server
           cabal
+          shellcheck
```

Comment:
[Seems that actionlint is a available in nixpkgs](https://github.com/rhysd/actionlint/blob/main/docs/install.md#nix), so maybe it would be better to add it here instead of manual download step in a workflow?

Reply:
Hum yeah I had missed that `actionlint` is in `nixpkgs` :facepalm:

I added it to the dev shell and removed the custom installation from the workflow :+1: